doc/manual/installation.md

```diff
@@ -0,0 +1,341 @@
+# Installation from source
+
+
+## Important Notes
+
+This guide is long because it covers many cases and includes all commands you need.
+
+This installation guide was created for and tested on **Debian/Ubuntu** operating systems. Please read [doc/install/requirements.md](./requirements.md) for hardware and operating system requirements.
+
+This is the official installation guide to set up a production server. To set up a **development installation** or for many other installation options please see [the getting started section of the readme](https://github.com/cantino/huginn#getting-started).
+
+The following steps have been known to work. Please **use caution when you deviate** from this guide. Make sure you don't violate any assumptions Huginn makes about its environment. For example many people run into permission problems because they change the location of directories or run services as the wrong user.
+
+If you find a bug/error in this guide please **submit a merge request**.
+
+If not stated otherwise all commands should be run as user with sudo permissions or as root.
+
+## Overview
+
+The Huginn installation consists of setting up the following components:
+
+1. Packages / Dependencies
+1. Ruby
+1. System Users
+1. Database
+1. Huginn
+1. Nginx
+
+## 1. Packages / Dependencies
+
+`sudo` is not installed on Debian by default. Make sure your system is
+up-to-date and install it.
+
+    # run as root!
+    apt-get update -y
+    apt-get upgrade -y
+    apt-get install sudo -y
+
+**Note:** During this installation some files will need to be edited manually. If you are familiar with vim set it as default editor with the commands below. If you are not familiar with vim please skip this and keep using the default editor.
+
+    # Install vim and set as default editor
+    sudo apt-get install -y vim
+    sudo update-alternatives --set editor /usr/bin/vim.basic
+
+Import node.js repository (can be skipped on Ubuntu and Debian Jessie):
+
+    curl -sL https://deb.nodesource.com/setup_0.12 | sudo bash -
+
+Install the required packages (needed to compile Ruby and native extensions to Ruby gems):
+
+    sudo apt-get install -y runit build-essential git zlib1g-dev libyaml-dev libssl-dev libgdbm-dev libreadline-dev libncurses5-dev libffi-dev curl openssh-server checkinstall libxml2-dev libxslt-dev libcurl4-openssl-dev libicu-dev logrotate python-docutils pkg-config cmake nodejs graphviz
+
+
+## 2. Ruby
+
+
+The use of Ruby version managers such as [RVM](http://rvm.io/), [rbenv](https://github.com/sstephenson/rbenv) or [chruby](https://github.com/postmodern/chruby) with Huginn in production frequently leads to hard-to-diagnose problems. Version managers are not supported and we strongly advise everyone to follow the instructions below to use a system Ruby.
+
```

Comment:
Yeah me too, I think huginn does not have the same problems GitLab has with rvm/rbenv, but getting the installation right with it is harder and can give strange errors especially when sudoing.